### DISCUSSION TITLE
How to use correlation threshold argument to avoid covariate dropping

### DISCUSSION THREAD

**Discussion by thomasklausch2** [amices/mice#366] [Missing data methodology]
When the correlation between target variable and a covariate is too high, the covariate is dropped and the event is logged, e.g.

```
set.seed(2021)
n = 1e3
x = rnorm(n)
y = x + rnorm(n,0,.1)
cor(y,x)
w = rbinom(n,1, 1/(1+exp(-2-x)) )
mean(w)
y[w==0] = NA
cor(y,x,use = 'complete.obs')
library(mice)
m = mice(data.frame(x,y), m = 5, maxit = 10)
head(m$loggedEvents)
```

In section 9.1.5 of your book (https://stefvanbuuren.name/fimd/sec-toomany.html) it is specified that this happens if the correlation is larger than .999. However in my example the correlation is just below this threshold, still the variable `x` gets dropped. Furthermore it is stated in the same section that the `threshold` argument can be used to increase this threshold. However, this arguments seems not to be working in the way I expect, as it does 'nothing'.

`m = mice(data.frame(x,y), m = 5, maxit = 10, threshold = .9999)`

Could you clarify how this setting is handled?

**Comment by stefvanbuuren** [comment]
The `threshold` parameter feeds into `find.collinear()` defined in <https://github.com/amices/mice/blob/master/R/internal.R>. This function is run once at initialisation. 

The related `remove.lindep()` function is run before each imputation in the sampler, and uses the `maxcor` parameter, which is `0.99` by default. 

Hope this help.

**Comment by stefvanbuuren** [comment]
Related: #225, #234, #278, #150

**Comment by thomasklausch2** [comment]
Thank you. Would it be possible that you show for my example how to change the `threshold`?  I have now tried `threshold=1`, `max.cor = 1` and `remove.collinear = F` - none of these work in the example.

**Comment by stefvanbuuren** [comment]
Did you try `maxcor`?

**Comment by thomasklausch2** [comment]
OK solved. Thanks!
